python/lsst/ts/salobj/topics/controller_event.py

```diff
@@ -58,34 +56,34 @@ def put(self, data=None, priority=1):
         if retcode != self.salinfo.lib.SAL__OK:
             raise RuntimeError(f"salProcessor({self.name}) failed with return code {retcode}")
 
-    def set_put(self, **kwargs):
+    def set_put(self, force_output=False, **kwargs):
         """Set one or more fields of ``self.data`` and put *if changed*.
 
-        The data is put if it has never been set (`has_data` False)
-        or if this call changes any field.
+        The data is put if it has never been set (`has_data` False), or this
+        call changes the value of any field, or ``force_output`` is true.
+
+        Parameters
+        ----------
+        force_output : `bool` (optional)
+            If True the output even if not changed.
```

Comment:
"If True, show the output even if not changed"?

Reply:
Agreed. I actually meant to write "If True then output even if not changed." but that might be a bit terse. I will change it to "If True then output the event, even if not changed".